### PR TITLE
[PORT] Coffee maker upgrade

### PR DESCRIPTION
**Pull request by QuacksQ**
## About The Pull Request

https://github.com/Artea-Station/Artea-Station-Server/assets/79924768/05cf7e30-864e-4a36-a8a3-27b6c772c058

This PR adds the much needed Impressa coffee maker from TG because its way better than the normal coffee maker we currently have.

This pr is a port of:
- https://github.com/tgstation/tgstation/pull/70991

## How Does This Help ***Gameplay***?
You can now drink coffee from a fancy coffee maker like a true space hipster.

## How Does This Help ***Roleplay***?
Better immersion in the coffee RP because we have a better coffee maker now.

## Proof of Testing
<details>
<summary>Screenshots/Videos</summary> 

https://github.com/Artea-Station/Artea-Station-Server/assets/79924768/87635e4d-9b97-4cd9-a807-70262ac27aba

</details>

## Changelog
:cl:
add: Adds a new and fancier coffee maker!
/:cl:

### CODE REVIEW

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 1]
```diff
@@ -1361,3 +1361,15 @@
 		/obj/item/stock_parts/capacitor = 1,
 		/obj/item/stock_parts/micro_laser = 1,
 	)
+
+/obj/item/circuitboard/machine/coffeemaker/impressa
+	name = "Impressa Coffeemaker"
+	greyscale_colors = CIRCUIT_COLOR_SERVICE
+	build_path = /obj/machinery/coffeemaker/impressa
+	req_components = list(
```
Comment (RimiNosha) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm:1369]: This var plus `greyscale_colors` are redundant.

**Review by RimiNosha — CHANGES_REQUESTED** [review]
I hate the amount of balloon alerts used, but that's a personal preference thing.

**Comment by RimiNosha** [comment]
This'll be good to merge once that review is addressed.

File: code/game/objects/items/circuitboards/machine_circuitboards.dm (hunk 1 above)
Comment (QuacksQ) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm:1367]: Adressed the review in latest commit

**Review by RimiNosha — APPROVED** [review]
Looks good

**Pull request merged**